### PR TITLE
[ticket/12559] Add forum setting to limit subforum legend

### PR DESCRIPTION
**Pull request by marc1706**
Checklist:

- [x] Correct branch: master for new features; 3.3.x & 3.2.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master](https://area51.phpbb.com/docs/dev/master/development/coding_guidelines.html), [3.3.x](https://area51.phpbb.com/docs/dev/3.3.x/development/coding_guidelines.html) and [3.2.x](https://area51.phpbb.com/docs/dev/3.2.x/development/coding_guidelines.html)
- [x] Commit follows commit message [format](https://area51.phpbb.com/docs/dev/3.3.x/development/git.html)

Tracker ticket (set the ticket ID to **your ticket ID**):

https://tracker.phpbb.com/browse/PHPBB3-12559

### CODE REVIEW

File: phpBB/includes/acp/acp_forums.php [hunk 1]
```diff
@@ -676,6 +678,7 @@ function main($id, $mode)
 					'S_ENABLE_INDEXING'			=> ($forum_data['enable_indexing']) ? true : false,
 					'S_TOPIC_ICONS'				=> ($forum_data['enable_icons']) ? true : false,
 					'S_DISPLAY_SUBFORUM_LIST'	=> ($forum_data['display_subforum_list']) ? true : false,
+					'S_DISPLAY_SUBFORUM_LIMIT'	=> ($forum_data['display_subforum_limit']) ? true : false,
 					'S_DISPLAY_ON_INDEX'		=> ($forum_data['display_on_index']) ? true : false,
 					'S_PRUNE_ENABLE'			=> ($forum_data['enable_prune']) ? true : false,
 					'S_PRUNE_SHADOW_ENABLE'			=> ($forum_data['enable_shadow_prune']) ? true : false,
```
Comment (rxu) [inline phpBB/includes/acp/acp_forums.php]: Not straightly related but probably a chance to change this kind of stuff to something like``` (bool) $forum_data['some_forum_data']``` 😛

File: phpBB/adm/style/acp_forums.html [hunk 2]
```diff
@@ -210,6 +210,11 @@ <h3>{L_WARNING}</h3>
 			<dd><label><input type="radio" class="radio" name="display_subforum_list" value="1"<!-- IF S_DISPLAY_SUBFORUM_LIST --> id="display_subforum_list" checked="checked"<!-- ENDIF --> /> {L_YES}</label>
 				<label><input type="radio" class="radio" name="display_subforum_list" value="0"<!-- IF not S_DISPLAY_SUBFORUM_LIST --> id="display_subforum_list" checked="checked"<!-- ENDIF --> /> {L_NO}</label></dd>
 		</dl>
+		<dl>
+			<dt><label for="display_subforum_limit">{L_LIMIT_SUBFORUMS}{L_COLON}</label><br /><span>{L_LIMIT_SUBFORUMS_EXPLAIN}</span></dt>
```
Comment (mrgoldy) [inline phpBB/adm/style/acp_forums.html]: Aren't we using twig syntax in all new added or amended  template code?

File: phpBB/phpbb/db/migration/data/v330/forums_legend_limit.php [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+class forums_legend_limit extends \phpbb\db\migration\migration
+{
+	public function effectively_installed()
+	{
+		return $this->db_tools->sql_column_exists($this->table_prefix . 'forums', 'display_subforum_limit');
+	}
+
+	static public function depends_on()
+	{
+		return array('\phpbb\db\migration\data\v330\v330b1');
```
Comment (mrgoldy) [inline phpBB/phpbb/db/migration/data/v330/forums_legend_limit.php]: Perhaps use new array syntax?

File: phpBB/includes/acp/acp_forums.php (hunk 1 above)
Comment (marc1706) [reply phpBB/includes/acp/acp_forums.php]: If I start here then that's going to a lot of refactoring. I do agree but also think this PR is not the place for that.

File: phpBB/adm/style/acp_forums.html (hunk 2 above)
Comment (marc1706) [reply phpBB/adm/style/acp_forums.html]: To be honest, I think this would look out of place here.

**Pull request merged**